### PR TITLE
get travis to run ngui tests

### PR DESCRIPTION
**Pull request by gaarf**
just trying

### CODE REVIEW

**Comment by gaarf** [comment]
if it worked, this would fix #536

**Comment by chtyim** [comment]
You can merge if the Travis passes on this PR. :)

**Comment by wolf31o2** [comment]
Don't remove the `ngui/.travis.yml` file, it may be useful, later...

File: .travis.yml [hunk 1]
```diff
@@ -25,3 +25,15 @@ branches:
     - /^feature\/.*$/
     - /^hotfix\/.*$/
     - /^release\/.*$/
+
+before_install:
+  - "export DISPLAY=:99.0"
+  - "sh -e /etc/init.d/xvfb start"
+
+before_script:
+  - "cd ngui && npm install"
```
Comment (wolf31o2) [inline .travis.yml]: After this, the current working directory is `ngui` so the next commands will fail...

**Comment by gaarf** [comment]
It works!
@chtyim @wolf31o2 please review :8ball:

**Comment by gaarf** [comment]
unit

File: ngui/test/karma.conf.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = function(config){
-  config.set({
+  var it = {
```
Comment (ajainarayanan) [inline ngui/test/karma.conf.js]: `it.should.be...` is generally used as an assertive variable in js testing. Is it possible to use some other name?

Comment (gaarf) [inline ngui/test/karma.conf.js]: sure, good call.

**Comment by ajainarayanan** [comment]
:+1: Looks good! Great to see travis being added! Time to churn unit tests in js :)

File: ngui/.travis.yml [hunk 3]
```diff
@@ -3,11 +3,11 @@ language: node_js
...
 
-before_script:
-  - export DISPLAY=:99.0
-  - sh -e /etc/init.d/xvfb start
-  - npm install
+before_install:
+  - "export CHROME_BIN=chromium-browser"
+  - "export DISPLAY=:99.0"
+  - "sh -e /etc/init.d/xvfb start"
+  - "npm run build"
```
Comment (wolf31o2) [inline ngui/.travis.yml]: Should we have this in the top-level `.travis.yml` also?

Comment (gaarf) [inline ngui/.travis.yml]: `npm run travis` includes the build step now, so I'll remove it from here.

**Comment by wolf31o2** [comment]
If this passes, you should go ahead and merge it. Make any further changes on another branch, since this has already been code reviewed.

**Pull request merged**